File: C++/650.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    int minSteps(int n) {
	int step = 0;
	if (n == 1)return 0;
	if (n == 2)return 2;
	while (true) {
		int t = get_init(n);
		if (t == -1) {
			step += n;
			break;
		}
		else {
			int z = n / t;
			if (z == 1) {
				step += 2;
				break;
			}
			if (z == 2) {
				step += 2;
			}
			else {
				step = step + z;
			}
			n = t;
		}
	}
	return step;
}
    private:
   int get_init(int n) {
	for (int i = n/2+1; i >=2; i--) {
		if (n % i == 0) {
			return i;
		}
	}
	return -1;
    }
};
__________________________________________________________________________________________________
sample 8720 kb submission
class Solution {
public:
    int minSteps(int n) {
        if (n <= 1)
            return 0;
        for (int i = 2; i <= n; i++) {
            if (n % i == 0) {
                return i + minSteps(n/i);
            }
        }
        return 0;
    }
};
__________________________________________________________________________________________________

File: Python3/650.py
__________________________________________________________________________________________________
sample 28 ms submission
import math

class Solution:
    def minSteps(self, n: int) -> int:
        if n == 1:
            return 0
        if n == 2:
            return 2
        
        def gcd(num):
            greatest = 1
            
            for i in range(2, num // 2):
                if num % i == 0:
                    greatest = i
                    
            return greatest
        
        def rec(num):
            if gcd(num) == 1:
                return num
            
            g = gcd(num)
            got = rec(g)
            return got + num // g
        
        return rec(n)
__________________________________________________________________________________________________
sample 12972 kb submission
class Solution:
    def minSteps(self, n: int) -> int:
        ans = 0
        f = 2
        while n > 1:
            while n % f == 0:
                ans += f
                n /= f
            f += 1
        return ans
__________________________________________________________________________________________________
